drgn_tools/crash_net.py

```diff
@@ -0,0 +1,368 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+import argparse
+import ipaddress
+import socket as socketlib
+
+from drgn import Program
+from drgn.helpers.linux import net as network
+from drgn.helpers.linux.list import hlist_for_each_entry
+from drgn.helpers.linux.net import SOCKET_I
+from drgn.helpers.linux.pid import find_task
+
+from drgn_tools.corelens import CorelensModule
+
+_NETDEV_HASHBITS = 8
+_NETDEV_HASHENTRIES = 1 << _NETDEV_HASHBITS
+NEIGH_NUM_HASH_RND = 4
+
+sockfamily_dict = {
+    "0": "AF_UNSPEC",
+    "1": "AF_UNIX",
+    "2": "AF_INET",
+    "10": "AF_INET6",
+    "16": "AF_NETLINK/ROUTE",
+}
+
+socktype_dict = {
+    "1": "SOCK_STREAM",
+    "2": "SOCK_DGRAM",
+    "3": "SOCK_RAW",
+    "4": "SOCK_RDM",
+    "5": "SOCK_SEQPACKET",
+    "6": "SOCK_DCCP",
+    "10": "SOCK_PACKET",
+}
+
+hwtype_dict = {
+    "0": "ARPHRD_NETROM",
+    "1": "ARPHRD_ETHER",
+    "2": "ARPHRD_EETHER",
+    "3": "ARPHRD_AX25",
+    "4": "ARPHRD_PRONET",
+    "5": "ARPHRD_CHAOS",
+    "6": "ARPHRD_IEEE802",
+    "7": "ARPHRD_ARCNET",
+    "8": "ARPHRD_APPLETLK",
+    "15": "ARPHRD_DLCI",
+    "19": "ARPHRD_ATM",
+    "23": "ARPHRD_METRICOM",
+    "24": "ARPHRD_IEEE1394",
+    "27": "ARPHRD_EUI64",
+    "32": "ARPHRD_INFINIBAND",
+}
+
+state_dict = {
+    "1": "NUD_INCOMPLETE",
+    "2": "NUD_REACHABLE",
+    "4": "NUD_STALE",
+    "8": "NUD_DELAY",
+    "16": "NUD_PROBE",
+    "32": "NUD_FAILED",
+    "64": "NUD_NOARP",
+    "128": "NUD_PERMANENT",
+}
```

Comment:
Oh, as mentioned in a lot of comments below, you'd probably be well-served to simply remove the prefixes from all the comments. EG, instead of listing `AF_INET`, you could just list `INET`, and avoid some of the odd string manipulation you're currently doing to remove the prefix later.